integration-libs/cdc/root/service/cdc-js.service.ts

```diff
@@ -297,19 +324,42 @@ export class CdcJsService implements OnDestroy {
         {
           key: 'httpHandlers.badRequestPleaseLoginAgain',
           params: {
-            errorMessage: response.statusMessage,
+            errorMessage: response.errorDetails,
           },
         },
         GlobalMessageType.MSG_TYPE_ERROR
       );
     }
   }
 
+
+  protected getSessionExpirationValue(): number {
+    if (this.cdcConfig?.cdc !== undefined) {
+      const filteredConfigs: any = this.cdcConfig.cdc.filter(
+        (conf) => conf.baseSite === this.getCurrentBaseSite()
+      );
+      if (filteredConfigs && filteredConfigs.length > 0) {
+        return filteredConfigs[0].sessionExpiration;
+      }
+    }
+    // Return a default value
+    return 3600;
+  }
+
+  private getCurrentBaseSite(): string {
+    let baseSite: string = '';
+    this.baseSiteService
+      .getActive()
+      .pipe(take(1))
+      .subscribe((data) => (baseSite = data));
+    return baseSite;
```

Comment:
This method and `getSessionExpirationValue` should return an `Observable` and `loginUserWithoutScreenSet` should `switchMap` it.

Reply:
noted.